### PR TITLE
TS-4868: Add configs back and cleanup other cluster init

### PR DESCRIPTION
**Pull request by PSUdaemon**

### CODE REVIEW

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/811/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/707/ for details.

File: cmd/traffic_cop/traffic_cop.cc [hunk 1]
```diff
@@ -657,9 +656,6 @@ config_reload_records()
   config_read_int("proxy.config.cop.source_port", &source_port, true);
 #endif
 
-  config_read_int("proxy.local.cluster.type", &tmp_int);
-  cluster_type = static_cast<MgmtClusterType>(tmp_int);
-
```
Comment (jpeach) [inline cmd/traffic_cop/traffic_cop.cc]: Remove ``cluster_type`` altogether?

File: mgmt/RecordsConfig.cc [hunk 2]
```diff
@@ -834,6 +860,61 @@ static const RecordElement RecordsConfig[] =
...
+  {RECT_CONFIG, "proxy.config.cluster.load_compute_interval_msecs", RECD_INT, "5000", RECU_NULL, RR_NULL, RECC_NULL, NULL, RECA_NULL}
+  ,
+  {RECT_CONFIG, "proxy.config.cluster.periodic_timer_interval_msecs", RECD_INT, "100", RECU_NULL, RR_NULL, RECC_NULL, NULL, RECA_NULL}
+  ,
+  {RECT_CONFIG, "proxy.config.cluster.ping_history_buf_length", RECD_INT, "120", RECU_NULL, RR_NULL, RECC_NULL, NULL, RECA_NULL}
+  ,
+  {RECT_CONFIG, "proxy.config.cluster.cluster_load_clear_duration", RECD_INT, "24", RECU_NULL, RR_NULL, RECC_NULL, NULL, RECA_NULL}
+  ,
+  {RECT_CONFIG, "proxy.config.cluster.cluster_load_exceed_duration", RECD_INT, "4", RECU_NULL, RR_NULL, RECC_NULL, NULL, RECA_NULL}
+  ,
```
Comment (jpeach) [inline mgmt/RecordsConfig.cc]: Is it really necessary to add these back if there's no way to enable clustering?

File: cmd/traffic_cop/traffic_cop.cc (hunk 1 above)
Comment (PSUdaemon) [inline cmd/traffic_cop/traffic_cop.cc]: There is another patch that I have as a WIP that is for totally removing it. It touches more things than it first appears.

File: mgmt/RecordsConfig.cc (hunk 2 above)
Comment (PSUdaemon) [inline mgmt/RecordsConfig.cc]: This is the problem that we ran into in the first place. We'd have to go through the code and match the defaults to the case if the config doesn't exist, etc.

**Comment by PSUdaemon** [comment]
@jpeach see #966.

**Comment by gtenev** [comment]
I have not looked enough to say if this change is necessary or not but just wanted to mention that it helped me with a crash I got after syncing my fork.

```
Starting program: /opt/apache/trafficserver/bin/traffic_manager
[Thread debugging using libthread_db enabled]
[E. Mgmt] log ==> [TrafficManager] using root directory '/opt/apache/trafficserver'
[New Thread 0x7ffff709b700 (LWP 24130)]
[New Thread 0x7fffee69a700 (LWP 24131)]

Program received signal SIGSEGV, Segmentation fault.
inet_network (cp=0x0) at inet_net.c:55
55     		if (*cp == '0')
Missing separate debuginfos, use: debuginfo-install libunwind-1.1-3.el6.x86_64 tcl-8.5.7-6.el6.x86_64
(gdb) where
#0  inet_network (cp=0x0) at inet_net.c:55
#1  0x0000000000430273 in main (argc=<value optimized out>, argv=<value optimized out>) at traffic_manager.cc:668
```

Applying the patch seemed to fix the problem for me.

**Comment by dragon512** [comment]
Fixes the issue I reported on

**Comment by dragon512** [comment]
+1

File: mgmt/RecordsConfig.cc (hunk 2 above)
Comment (bryancall) [inline mgmt/RecordsConfig.cc]: Why do the configs need to be added back in?

Comment (PSUdaemon) [inline mgmt/RecordsConfig.cc]: There is a bunch of code that asserts on the existence. And the more we dig into it the more you have to remove, ala #966. This broke `traffic_cop` and `traffic_manager` without these configs.

**Pull request merged**